### PR TITLE
Setup CMake

### PR DESCRIPTION
**Pull request by esclapez**
to either compile PelePhysics as a stand-alone library or begin able to add PelePhysics to a project with:

<Pass PelePhysics options>
add_subdirectory(${PELE_PHYSICS_LOCATION})

target_link_libraries(myApp PUBLIC PelePhysics::${PELEPHYSICS_LIB}_api)

The lib name depends on the CHEM/EOS/TRANSPORT model it was compiled with.

### CODE REVIEW

**Comment by esclapez** [comment]
@jrood-nrel I am fairly clueless when it comes to CMake, so hopefully this can still be useful. Tested the stand alone build of the library with CUDA and HIP successfully, and the include with PeleLMeX with plain MPI so far.

**Comment by jrood-nrel** [comment]
I think this is a good idea. I can try to test it out.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,284 @@
+cmake_minimum_required(VERSION 3.14)
```
Comment (jrood-nrel) [inline CMakeLists.txt]: I would set this to 3.23. Even though I think HIP support exists in lower versions, other projects are requiring 3.23 now.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    HOMEPAGE_URL "https://github.com/AMReX-Combustion/PelePhysics"
+    LANGUAGES CXX
+    )
+
+message(STATUS "CMake version: ${CMAKE_VERSION}")
+
+#
+# Check if CMAKE_BUILD_TYPE is given. If not, use default
+#
+if ( NOT CMAKE_BUILD_TYPE )
```
Comment (jrood-nrel) [inline CMakeLists.txt]: I would remove this whole if statement. It's best to let CMake handle `CMAKE_BUILD_TYPE` itself.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+endif ()
+
+
+#
+# Define the object library to compile
+#
+set(pelephysics_lib_name PelePhysics-${PELEPHYSICS_CHEMISTRY_MODEL}-${PELEPHYSICS_EOS_MODEL}-${PELEPHYSICS_TRANSPORT_MODEL})
+add_library(${pelephysics_lib_name} OBJECT)
+if (BUILD_SHARED_LIBS)
+  set_target_properties(${pelephysics_lib_name} PROPERTIES POSITION_INDEPENDENT_CODE ON)
```
Comment (jrood-nrel) [inline CMakeLists.txt]: I never really know which way to go with this. Sometimes it's good to always have PIC on, but I think it's best to start by letting CMake handle it by removing this.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+endif ()
+
+#
+# Enable CUDA if requested
+#
+if (PELEPHYSICS_GPU_BACKEND STREQUAL "CUDA")
+   # CMake 3.18+: CMAKE_CUDA_ARCHITECTURES
+   # https://cmake.org/cmake/help/latest/policy/CMP0104.html
+   if(POLICY CMP0104)
```
Comment (jrood-nrel) [inline CMakeLists.txt]: I think we can remove this if setting the CMake minimum version higher.

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+
+message( STATUS "PelePhysics configuration summary: ")
+message( STATUS "   Build type               = ${CMAKE_BUILD_TYPE}")
+message( STATUS "   Install directory        = ${CMAKE_INSTALL_PREFIX}")
+message( STATUS "   C++ compiler             = ${CMAKE_CXX_COMPILER}")
+message( STATUS "   C++ include paths        = ${_cxx_includes}")
+message( STATUS "   CMake target             = ${pelephysics_lib_name}_api")
+
+# Create non-object library for use as external target
+add_library(${pelephysics_lib_name}_api STATIC)
```
Comment (jrood-nrel) [inline CMakeLists.txt]: I wouldn't choose `STATIC` or `SHARED` here and let CMake decide.

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**